### PR TITLE
[MM-37445] Create config setting to always have the team sidebar visible

### PR DESCRIPTION
**Pull request by kyeongsoosoo**
#### Summary

Added the option to always show team sidebar in the System console.

I resubmit the PR because the ticket has been reopened.
Original PR: https://github.com/mattermost/mattermost/pull/23402

#### Ticket Link

Closes https://github.com/mattermost/mattermost/issues/18010
https://mattermost.atlassian.net/browse/MM-37445

#### Release Note

```release-note
Create config setting to always have the team sidebar visible
```

### CODE REVIEW

**Comment by kyeongsoosoo** [comment]
@mickmister I reopned the PR for the ticket https://github.com/mattermost/mattermost/issues/18010

File: server/public/model/config.go [hunk 1]
```diff
@@ -256,6 +256,8 @@ const (
 	OpenidSettingsDefaultScope = "profile openid email"
 
 	LocalModeSocketPath = "/var/tmp/mattermost_local.socket"
+
+	AlwaysShowTeamSidebar = ""
```
Comment (mickmister) [inline server/public/model/config.go:260]: Is this value ever referenced?

**Review by mickmister — APPROVED** [review]
LGTM, thanks @kyeongsoosoo 🚀

**Comment by mickmister** [comment]
@lindalumitchell Do you know who would be the right QA reviewer to take a look at this?

File: server/public/model/config.go (hunk 1 above)
Comment (calebroseland) [reply server/public/model/config.go:260]: +1

**Review by calebroseland — APPROVED** [review]

**Comment by lindalumitchell** [comment]
@mickmister @lindy65 is on it. :)

**Review by lindy65 — APPROVED** [review]
Thanks @kyeongsoosoo - looks good to me 👍 

@mickmister - please run E2E tests - thanks!

File: server/public/model/config.go (hunk 1 above)
Comment (kyeongsoosoo) [reply server/public/model/config.go]: Thanks for the catch. 
This value had no use. I removed it.

**Comment by mickmister** [comment]
Tests are all passing except for some enterprise tests that are unrelated. @kyeongsoosoo I'm thinking this will be resolved within a few days. Thanks so much for your effort on this!

**Comment by hanzei** [comment]
@kyeongsoosoo Heads up that your changes broke the E2E tests: `Error: support/server/default_config.ts(80,5): error TS2741: Property 'AlwaysShowTeamSidebar' is missing in type '{ SiteURL: string; WebsocketURL: string; LicenseFileLocation: string; ListenAddress: string; ConnectionSecurity: string; TLSCertFile: string; TLSKeyFile: string; TLSMinVer: string; TLSStrictTransport: false; ... 101 more ...; MaximumPayloadSizeBytes: number; }' but required in type 'ServiceSettings'.`

Can you please take a look?

**Comment by hanzei** [comment]
I'm going to close the PR due to inactivity. @kyeongsoosoo please let me know if you still want to work on the issue. Thanks!

**Pull request closed without merging**